### ISSUE TITLE
what to do next question

### ISSUE THREAD

**Issue by jkoschinsky** [GeoDaCenter/reactgeoda#144]
I loaded the [old stlouis file](https://geodacenter.github.io/data-and-lab/stlouis/) into webgeoda and asked a more open-ended question that a new user would ask: "I just uploaded my data. How should I get started with the analysis?"

It then gives the following answer:
<img width="325" alt="Screenshot 2024-07-01 at 11 14 24 AM" src="https://github.com/GeoDaCenter/reactgeoda/assets/13679743/8a1218ed-173f-4a46-9a94-bf0844d7dfd6">

but then this answer disappears and is replaced by the following, which doesn't make sense:

<img width="361" alt="Screenshot 2024-07-01 at 11 14 04 AM" src="https://github.com/GeoDaCenter/reactgeoda/assets/13679743/3c8de9c9-15e8-4b32-a07e-3b906ca432ed">

I ask it to list the existing variables in the spatial file and the answer confirms that it doesn't recognize the question:
<img width="352" alt="Screenshot 2024-07-01 at 11 15 45 AM" src="https://github.com/GeoDaCenter/reactgeoda/assets/13679743/35080462-3ddf-4de6-8282-d7bf508b6e82">

Can we add a prompt for the user to re-specify the question in a way that the bot can identify the existing variables?

**Comment by lixun910** [comment]
Good point! We used to have a function to tell the LLM the meta data like variable names and types (not the raw data). We disabled it because it is a little slow when sending the meta data to openai. With the boosted performance of the new gpt4o model, I think it is time to enable this function so the LLM knows the meta data like variable names etc. The issues like this should be solved then.

**Comment by lixun910** [comment]
But this could be a potential privacy issue that users may argue. We can make it optional later.

**Comment by jkoschinsky** [comment]
Thx! It would also be good to leave the original answer without overriding
it since that's what we can tweak for suggesting reasoning steps.

**Comment by lixun910** [comment]
Should be fixed. Please help to check, Thanks! :)
